File: examples/Nodejs/request_utf8.js
#!/usr/bin/env node

var scrapy = require('scrapystreaming');
var fs = require('fs');


scrapy.createSpider('utf8sample', [], function(response) {});
scrapy.sendRequest('http://httpbin.org/encoding/utf8', function(response) {
    fs.writeFile('outputs/utf8.html', response.body);
    scrapy.closeSpider();
});

scrapy.runSpider();

File: examples/Python/request_utf8.py
#! /usr/bin/env python
# coding: utf-8
from utils import *


def main():

    status = parse_json(stdin.readline())

    # we start checking if the channel is ready
    if status['status'] != 'ready':
        raise Exception("There is problem in the communication channel")

    write_line('''
        {
            "type": "spider",
            "name": "utf8",
            "start_urls": []
        }
        ''')

    write_line('''
            {
                "type": "request",
                "id": "page",
                "url": "http://httpbin.org/encoding/utf8"
            }''')
    
    response = stdin.readline()
    data = parse_json(response)

    with open('outputs/utf8.html', 'wb') as f:
        f.write(data['body'].encode('utf-8'))
    

if __name__ == '__main__':
    main()
